internal/provider/resource_aci_pim_route_map_entry.go

```diff
@@ -0,0 +1,595 @@
+	if TagAnnotationchildPayloads == nil {
+		return nil, errMessage, errMessageDetail
+	}
+	childPayloads = append(childPayloads, TagAnnotationchildPayloads...)
+
+	payloadMap["children"] = childPayloads
+	if !data.Action.IsNull() && !data.Action.IsUnknown() {
+		payloadMap["attributes"].(map[string]string)["action"] = data.Action.ValueString()
+	}
+	if !data.Annotation.IsNull() && !data.Annotation.IsUnknown() {
+		payloadMap["attributes"].(map[string]string)["annotation"] = data.Annotation.ValueString()
+	}
+	if !data.Descr.IsNull() && !data.Descr.IsUnknown() {
+		payloadMap["attributes"].(map[string]string)["descr"] = data.Descr.ValueString()
+	}
+	if !data.Grp.IsNull() && !data.Grp.IsUnknown() {
+		payloadMap["attributes"].(map[string]string)["grp"] = data.Grp.ValueString()
+	}
+	if !data.Name.IsNull() && !data.Name.IsUnknown() {
+		payloadMap["attributes"].(map[string]string)["name"] = data.Name.ValueString()
+	}
+	if !data.NameAlias.IsNull() && !data.NameAlias.IsUnknown() {
+		payloadMap["attributes"].(map[string]string)["nameAlias"] = data.NameAlias.ValueString()
+	}
+	if !data.Order.IsNull() && !data.Order.IsUnknown() {
+		payloadMap["attributes"].(map[string]string)["order"] = data.Order.ValueString()
+	}
+	if !data.Rp.IsNull() && !data.Rp.IsUnknown() {
+		payloadMap["attributes"].(map[string]string)["rp"] = data.Rp.ValueString()
+	}
+	if !data.Src.IsNull() && !data.Src.IsUnknown() {
+		payloadMap["attributes"].(map[string]string)["src"] = data.Src.ValueString()
+	}
+
+	payload, err := json.Marshal(map[string]interface{}{"pimRouteMapEntry": payloadMap})
+	if err != nil {
+		errMessage := "marshalling of json payload failed"
+		errMessageDetail := fmt.Sprintf("err: %s. Please report this issue to the provider developers.", err)
+		return nil, errMessage, errMessageDetail
+	}
+
+	jsonPayload, err := container.ParseJSON(payload)
+
+	if err != nil {
+		errMessage := "construction of json payload failed"
+		errMessageDetail := fmt.Sprintf("err: %s. Please report this issue to the provider developers.", err)
+		return nil, errMessage, errMessageDetail
+	}
+	return jsonPayload, "", ""
+}
+
+func getPimRouteMapEntryDeleteJsonPayload(ctx context.Context, data *PimRouteMapEntryResourceModel) (*container.Container, string, string) {
+
+	jsonString := fmt.Sprintf(`{"pimRouteMapEntry":{"attributes":{"dn": "%s","status": "deleted"}}}`, data.Id.ValueString())
+	jsonPayload, err := container.ParseJSON([]byte(jsonString))
+	if err != nil {
+		errMessage := "construction of json payload failed"
+		errMessageDetail := fmt.Sprintf("err: %s. Please report this issue to the provider developers.", err)
+		return nil, errMessage, errMessageDetail
+	}
+	return jsonPayload, "", ""
+}
+
+func doPimRouteMapEntryRequest(ctx context.Context, client *client.Client, path, method string, payload *container.Container) (*container.Container, string, string) {
```

Comment:
I kept it separated for trouble shooting, and if deviations are required for a specific class in future. 

Since it is auto-generated code I thought that it would not hurt defining also for every module. We could create a manual utils functions for these functions ( basically replacing the template part ).

Reply:
The only issue would be this -
If we decided to make a change to any of these functions that applies to all the resources, we would have to re-generate the existing resources again with the changes.

Reply:
Yes correct, any change would require full regenerate, is that a bad thing?

Reply:
Not sure as it depends on what we want. Re-generating test cases with additional tests makes sense though.

Reply:
to be done at later stage